### PR TITLE
allow to send many messages on single connection

### PR DESCRIPTION
**Pull request by ronenhamias**
allow to send many messages on single connection (while keeping prev behavior in parallel)

example message:
```json
q = qualifier
d = data (json object)
{
  "q":"/greeting/one",
  "d":"5"
}
```

### CODE REVIEW

File: Websocket/src/main/java/io/scalecube/gateway/websocket/JsonCodec.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  public ServiceMessage decode(ByteBuf content) throws Exception {
+
+    byte[] bytes = new byte[content.readableBytes()];
+    int readerIndex = content.readerIndex();
+    content.getBytes(readerIndex, bytes);
+    HashMap frame = mapper.readValue(bytes, HashMap.class);
+    Builder builder = ServiceMessage.builder()
+        .header("content-type", APPLICATION_JSON);
+
+    if (frame.containsKey("d")) {
```
Comment (segabriel) [inline Websocket/src/main/java/io/scalecube/gateway/websocket/JsonCodec.java]: move "d" to some constants

File: Websocket/src/main/java/io/scalecube/gateway/websocket/WebSocketSession.java [hunk 2]
```diff
@@ -115,16 +131,45 @@ public String contentType() {
...
+    };
+  }
+
+  private Function<WebSocketFrame, ServiceMessage> decodeMessage() {
+    return (WebSocketFrame frame) -> {
+      try {
+        return codec.decode(frame.content());
+      } catch (Exception e) {
+        return ServiceMessage.builder()
+            .qualifier("error")
```
Comment (segabriel) [inline Websocket/src/main/java/io/scalecube/gateway/websocket/WebSocketSession.java]: use Qualifier.asError() for it

File: Websocket/src/test/java/io/scalecube/gateway/websocket/WebSocketResource.java [hunk 3]
```diff
@@ -56,21 +57,23 @@ public InetSocketAddress newServer(Function<WebSocketSession, Mono<Void>> onConn
...
   }
 
-  public <T> Flux<T> newClientSession(String path, List<T> requests, Class<T> clazz) {
+  
+  public <T> Flux<HashMap> newClientSession(String path, List<T> req, Class<HashMap> class1) {
 
     WebSocketClient client = new ReactorNettyWebSocketClient();
     URI uri = getUri(path);
-    ReplayProcessor<T> responses = ReplayProcessor.create();
+    ReplayProcessor<HashMap> responses = ReplayProcessor.create();
```
Comment (segabriel) [inline Websocket/src/test/java/io/scalecube/gateway/websocket/WebSocketResource.java]: why don't just generic Map use?

File: Websocket/src/main/java/io/scalecube/gateway/websocket/WebSocketSession.java [hunk 4]
```diff
@@ -115,16 +131,45 @@ public String contentType() {
...
+            .dataFormat(contentType)
+            .data(e.getMessage())
```
Comment (segabriel) [inline Websocket/src/main/java/io/scalecube/gateway/websocket/WebSocketSession.java]: We should use errorData for it

**Comment by ronenhamias** [comment]
Done in another pull request

**Pull request closed without merging**